fk/secretreceive.go

```diff
@@ -202,13 +230,77 @@ func countDigits(i int) (count int) {
 	return len(iString)
 }
 
+func getFileNameToWrite(filename string) (string, error) {
+	userDir, err := homedir.Dir()
+	if err != nil {
+		return "", err
+	}
+	downloadsDir := userDir + "/Downloads"
+	fileInfo, err := os.Stat(downloadsDir)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return "", fmt.Errorf(downloadsDir + " directory doesn't exist")
+		} else {
+			return "", err
+		}
+	}
+	if !fileInfo.IsDir() {
+		return "", fmt.Errorf(fileInfo.Name() + " file exists and is not a directory")
+	}
+	filenameToWrite := getNewUniqueFilename(
+		filepath.FromSlash(downloadsDir+"/"+filename),
+		doesntExist,
+	)
+
+	return filenameToWrite, nil
+}
+
+func getNewUniqueFilename(fp string, checker func(string) bool) string {
+	i := 0
+	foundFilename := false
+	filename := filepath.Base(fp)
+	numberedFilename := filename
+	for ok := true; ok; ok = (foundFilename == false) {
```

Comment:
I read a rule once from a reliability engineer that read something like "never let anything be unbounded / always limit the number of attempts". I think it applies here: if something strange went wrong and `foundFilename` kept refusing to be true (for example, because of some permissions thing), I think it's unlikely we should try more than, say, 10 different filenames. (We certainly wouldn't want to be trying millions e.g. `/tmp/secret21674364736743.txt`)

So I propose to use the simpler `for` form: `for suffix := 1;  suffix<=20; suffix++ {` and have the for loop return on success. If it falls out of the loop, we know something weird happened, and can return an error.

Reply:
Sensible.